src/query/sql/src/planner/binder/table.rs

```diff
@@ -320,7 +347,11 @@ impl<'a> Binder {
                     push_down_predicates: None,
                     limit: None,
                     order_by: None,
-                    statistics: stat,
+                    statistics: Statistics {
+                        statistics: stat,
+                        col_stats,
```

Comment:
`col_stats` is only used by `count(col)` aggregate function, it seems that there is no need for a binder to every table, and lazy loading seems to be a better choice.

Reply:
Good point

Reply:
I try to modify the code. And add it in post_optimize, but we should use the function 

```rust
table
    .column_statistics_provider()
    .await?
    .column_statistics(col_id as ColumnId);
col_stats.insert(column.index(), col_stat);

// because of in pub async fn snapshot_loc(&self) -> Result<Option<String>> : self.operator.object(url).read().await?;
```

If set post_optimize to async, It's too invasive.

So, I'm not sure, do we have other way to get the col_stats? cc @dantengsky  @lichuang

Reply:
> `col_stats` is only used by `count(col)` aggregate function, it seems that there is no need for a binder to every table, and lazy loading seems to be a better choice.

Actually we try to lazy loading this, but there has a question:

the statistics is saved in logicalget, now we can get a column_stat provider(thx @dantengsky ) but in post_optimize, the count() pattern type is :

```sql
pattern: SExpr::create_unary(
                PatternPlan {
                    plan_type: RelOp::Aggregate,
                }
                .into(),
                SExpr::create_leaf(
                    PatternPlan {
                        plan_type: RelOp::Pattern,
                    }
                    .into(),
                ),
            ),
```

So we can't downcast the relop::Pattern to logicalget. 

Because of this, I set this comment to resolved